xapian-core/weight/bm25weight.cc

```diff
@@ -35,11 +35,23 @@ using namespace std;
 
 namespace Xapian {
 
+BM25Weight::BM25Weight(bool& bm25+)
```

Comment:
I really don't mind, although right now I don't see how you can tune the other parameters while using BM25+, which suggests that this is the wrong API. An alternative would be to either have it set on the class (to `false`), and override to `true` on a `BM25PlusWeight` subclass, or to factor out the bits that need to change in the calculations, and override them.